audio-worklet/bitcrusher.js

```diff
@@ -0,0 +1,234 @@
+    return rounded / factor;
+  };
+  
+  /**
+    Connect this script processor node to target Audio Node
+    @param {AudioNode} target  
+    @return {Bitcrusher} this for chaining
+  */
+  connectOut(dest) {
+    this.node_.connect(dest) 
+    return this;
+  }
+
+  /** 
+    Connect input AudioNode to internal script processor node
+    @param {AudioNode}  
+    @return {Bitcrusher} this for chaining
+  */
+  connectIn(input) {
+    input.connect(this.node_);
+    return this; 
+  }
+  
+  /** 
+    Disconnect input AudioNode to internal script processor node
+    @param {AudioNode} dest to disconnect from
+    @return {Bitcrusher} this for chaining
+  */
+  disconnect (dest) {
+    this.node_.disconnect(dest);
+    return this; 
+  }
+}
+
+testBitcrusher();
+
+/**
+  Run series of tests on Bitcrusher node, 
+*/
+function testBitcrusher() {
+  /* ---------------------------------------------------------------------------- 
+     Test Bitcrusher rounding method. 
+     ----------------------------------------------------------------------------*/
+  let base = 10; // set base to 10 for easier testing
+  let x = Bitcrusher.round(1.7821, 2, base); 
+  let expected = 1.78;
+  console.assert(x == expected, "computed " + x + " but expected " + expected);
+
+  x = Bitcrusher.round(-1.332432, 5, base); 
+  expected = -1.33243;
+  console.assert(x == expected, "computed " + x + " but expected " + expected);
+
+  x = Bitcrusher.round(10.322, 3, base); 
+  expected = 10.322;
+  console.assert(x == expected, "computed " + x + " but expected " + expected);
+
+  x = Bitcrusher.round(23.432123212321232123, 9, base); 
+  expected = 23.432123212;
+  console.assert(x == expected, "computed " + x + " but expected " + expected);
+
+  x = Bitcrusher.round(1, 1, base); 
+  expected = 1;
+  console.assert(x == expected, "computed " + x + " but expected " + expected);
+  /* ---------------------------------------------------------------------------- */
+
+  
+  /* ---------------------------------------------------------------------------- 
+     Test sample reduction component. 
+     ---------------------------------------------------------------------------- */
+  // Hold constant precision and test factor reduction
+  // Populate a Float Array with 0:99 and verify the bitcrusher holds [0, 0, ... 0] [10, 10, ... 10]
+  // where the number of consecutive equal numbers is controlled by factor
+  for (let factor = 1; factor < 10; factor++) {
+    const precision = 1; // leave precision in tact
+    const bufferSize = 100; // arbitrarily sized buffer for testing
+    let referenceBuffer = new Float32Array(bufferSize);
+    let outputBuffer = new Float32Array(bufferSize);
+    
+    for (let i =0; i < bufferSize; i++) {
+      referenceBuffer[i] = i; // populate array 1:100
+    }
+
+    Bitcrusher.bitcrush(factor, precision, referenceBuffer, outputBuffer); 
+    
+    outputBuffer.map(function(d, i) { // d is bitcrushed data and i is index in array
```

Comment:
Use the arrow function notation, and expand the variable names. Then you don't need comments here.